themes/bootstrap5/js/common.js

```diff
@@ -103,6 +118,44 @@ var VuFind = (function VuFind() {
       fn(...args);
     }
   }
+  // Promises and resolves.
+
+  /**
+   * Create a promise for an event. Promises are useful in situations, where code requires something to be available
+   * before executing. I.e calling getPromise('document-ready').then(() => {...code}) will execute the function
+   * after VuFind module has called resolvePromise('document-ready'). After promise has been resolved, every
+   * call to getPromise('document-ready).then() will succeed.
+   * @param {string} identifier Identifier for the promise. Same identifier has to be used with resolvePromise.
+   * @returns {Promise} Promise which has to be resolved by using resolvePromise
+   */
+  function createPromise(identifier) {
+    if (!_promises[identifier]) {
+      _promises[identifier] = new Promise((resolve) => _resolves[identifier] = resolve);
+    }
+    return _promises[identifier];
+  }
+
+  /**
+   * Get a promise.
+   * @param {string} identifier Identifier for the promise.
+   * @returns {Promise|undefined} Promise or undefined.
+   */
+  function getPromise(identifier) {
+    return _promises[identifier] || undefined;
```

Comment:
4. I am concerned about this part of the implementation. This seems to introduce an issue where the listener (`getPromise`) is called before the emit (`createPromise` or `resolvePromise`). What if undefined is returned? I think we'd get an error similar to `(undefined).then(() => {})`.